storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -616,6 +656,18 @@ public void pruneHotStateRoots(final List<Bytes32> stateRoots) {
     return protoArrayDao.getProtoArraySnapshot();
   }
 
+  @Override
+  public void addNonCanonicalBlock(final SignedBeaconBlock block) {
+    if (!storeNonCanonicalBlocks) {
+      return;
+    }
+    try (final FinalizedUpdater updater = finalizedDao.finalizedUpdater()) {
+      LOG.debug("Non canonical block {}", block.getRoot().toHexString());
+      updater.addNonCanonicalBlock(block);
+      updater.commit();
+    }
+  }
```

Comment:
We don't want to use a new transaction for every block we store.  Suspect this method should be on `FinalizedUpdater` instead so it just adds them to the existing transaction and then they are committed with the rest of the changes.